### ISSUE TITLE
Feature request: get rid of "sync engine"/external server dependence

### ISSUE THREAD

**Issue by SeAlgoAsoma** [nylas/N1#2889]
Nylas sounds great, but I'm not looking at running my own server, nor am I interested in handing a 3rd party access all my email to "sync". I've ruled out numerous current email apps for pulling this sort of thing. An open-source email client (that is under active development and isn't the buggy, almost unusable Thunderbird) sounds great, but as it is, I can't even try it out to see if it's for me, because you expect me to hand over all my email to you for some strange reason that I can't quite understand, and that's never going to happen.

 For many years, the best email apps have not required "syncing" on an additional external server to work well. I don't see a single feature in N1 that I haven't seen in plugins for existing (closed-source) clients that don't have this burdensome dependency on having an account on an external server for "syncing" or anything else besides IMAP and SMTP. It's a shame that this was architected in such a poor way, we could really use a good email client on OS X, with all the current development trends seeming to lead in similarly poor, user-disempowering directions.

I've read the (to me) poor rationales for this decision already and don't need them re-explained; just voicing my request that this requirement be reconsidered.

Opening a new issue because I noticed the comment I left on this was on an already-closed issue.

**Comment by grinich** [comment]
Hi @SeAlgoAsoma,

What you're proposing here is an architectural shift in the entire app, and not a current bug or enhancement request for the project, so I'm going to close this issue.

There are a lot of benefits of decoupling sync from the mail app itself, including compatibility with more providers, maturity of libraries in non-JS languages, and features that require offline processing. (e.g. send later, snoozing, sharing, reminders, etc.)

You don't need to use our hosted sync system. You're welcome to run the sync engine yourself - there are instructions here:
https://github.com/nylas/N1/blob/master/CONFIGURATION.md and the codebase is here: https://github.com/nylas/sync-engine. It's also GPL.

Similar answers that might help:

https://github.com/nylas/N1/issues/35

https://github.com/nylas/sync-engine/issues/243

https://support.nylas.com/hc/en-us/articles/217518207-Why-does-N1-sync-email-via-the-cloud-

**Comment by SeAlgoAsoma** [comment]
Ouch. Too bad.

**Comment by SeAlgoAsoma** [comment]
Here's an idea, can you not bundle the Sync Engine with the app as a binary?

**Comment by SeAlgoAsoma** [comment]
(Does "closed" mean these additional comments aren't going to be seen now? Do I need to open a new issue to ask the above?)

**Comment by grinich** [comment]
Nope we can keep discussing it here (and I'll get notifications). Our team uses GitHub issues for our open bug list so we close issues like this one but are happy to keep chatting.

Bundling the Python sync engine with N1 is a really interesting idea-- I'd be interested in seeing someone fork the app and make it happen. It's a bit complicated because the sync engine is designed to run on Debian Linux with MySQL and lots of Python dependencies. Nothing about that is impossible though!

I totally understand your argument for not using a cloud sync system. It just has different tradeoffs which don't fit our goals of the Nylas N1 product. But the beauty of open source is that we welcome people who disagree to just click the fork button! 😄

**Comment by SeAlgoAsoma** [comment]
Well, yeah, and I really appreciate your willingness to chat about it - but the unfortunate truth is some of us own businesses and were really hoping to just find a usable email client, do our email, and get back to running the shop. If I had the time to learn software development I'd have written my own client by now (and believe me, I looked into it, I'm a database guy by trade and it didn't seem like that much of a stretch, at first...) 

The current field of options, at least on Mac, is awful... either inflexible and underpowered, or inflexible and singularly focused on arcane (to me) features that I'm sure some people find useful but I never needed or wanted, and absolutely nothing in-between. There's nothing for the users who may not work like an email app developer. For all the bugs and missing functionality, Eudora let me work how I wanted, didn't force me to work how it wanted me to work. But that design philosophy seems to be completely extinct now. There are a raft of very simple, obvious usability features that not a single current email client offers. So, an open source client seemed attractive. 

I'm glad you think there may be a path forward in bundling the engine. On OS X, at least, I believe Python and MySQL are native citizens. Personally, if I was a python developer & had the time, I'd take it on myself. 

It seems like a logical way forward... why enforce a divided architecture, requiring two separate machines, if there's a way not to? (Ignoring for a moment the fact that you now have an existing codebase that's already built that way which would have to be converted.)

I had actually taken a look at the Sync Engine stuff before I posted here, unfortunately it's all greek to me, far outside my expertise. I don't even understand the basic introductory page. I can't begin to even make head or tail of what I would need to do to even use it.  So between that and me having a business to run, I couldn't even dream of being the one to fork it and take the project on... but I'd like to put the idea out there in case anyone else is interested. 

So, let me ask you this, with this thread closed, if I were to make bundling the sync server with the it's own feature request, could that be left open for some interested party to take on? Or is there some other more appropriate place I could post it where someone in the community with the time & knowledge might see it and take an interest? Or is this thread, even closed, visible enough to serve as that?

Thanks!
